src/master-playlist-controller.js

```diff
@@ -793,6 +806,37 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
+   * Check if a playlist has stopped being updated
+   * @param {Object} playlist the media playlist object
+   * @return {boolean} whether the playlist has stopped being updated or not
+   */
+  stuckAtPlaylistEnd_(playlist) {
+    let seekable = this.seekable();
+
+    if (!seekable.length) {
+      // playlist doesn't have enough information to determine whether we are stuck
+      return false;
+    }
+
+    // does not use the safe live end to calculate playlist end, since we
+    // don't want to say we are stuck while there is still content
+    let absolutePlaylistEnd = Hls.Playlist.playlistEnd(playlist);
+    let currentTime = this.tech_.currentTime();
+    let buffered = this.tech_.buffered();
+
+    if (!buffered.length) {
+      // playhead reached absolute end of playlist
```

Comment:
My fault, but maybe add a clarification (like you did below in the return comment 👍 ) to something like: `// return true if the playhead reached the absolute end of the playlist`